controller_manager:
  ros__parameters:
    update_rate: 30
    use_sim_time: true

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    joint_broad:
      type: joint_state_broadcaster/JointStateBroadcaster

joint_trajectory_controller:
  ros__parameters:
    joints: 
      - leg1_coxa
      - leg1_femur
      - leg1_tibia
      - leg2_coxa
      - leg2_femur
      - leg2_tibia
      - leg3_coxa
      - leg3_femur
      - leg3_tibia
      - leg4_coxa
      - leg4_femur
      - leg4_tibia
      - leg5_coxa
      - leg5_femur
      - leg5_tibia
      - leg6_coxa
      - leg6_femur
      - leg6_tibia

    command_interfaces:
      - position

    state_interfaces:
      - position
      - velocity

    # action_monitor_rate: 20.0

    # allow_partial_joints_goal: false
    # open_loop_control: true
    # constraints:
    #   stopped_velocity_tolerance: 0.01
    #   goal_time: 0.0
    #   joint1:
    #     trajectory: 0.05
    #     goal: 0.03